client/src/modules/fee_center/modals/fee_center.modal.js

```diff
@@ -0,0 +1,150 @@
+angular.module('bhima.controllers')
+  .controller('FeeCenterModalController', FeeCenterModalController);
+
+FeeCenterModalController.$inject = [
+  '$state', 'FeeCenterService', 'ModalService', 'NotifyService', 'appcache',
+];
+
+function FeeCenterModalController($state, FeeCenter, ModalService, Notify, AppCache) {
+  const vm = this;
+  vm.feeCenter = {};
+  vm.referenceFeeCenter = [];
+
+  const cache = AppCache('FeeCenterModal');
+
+  if ($state.params.creating || $state.params.id) {
+    cache.stateParams = $state.params;
+    vm.stateParams = cache.stateParams;
+  } else {
+    vm.stateParams = cache.stateParams;
+  }
+  vm.isCreating = vm.stateParams.creating;
+
+  // exposed methods
+  vm.submit = submit;
+  vm.auxiliaryFee = auxiliaryFee;
+  vm.costCenter = costCenter;
+  vm.onSelectAccountReference = onSelectAccountReference;
+  vm.onServicesChange = onServicesChange;
+
+  if (!vm.isCreating) {
+    FeeCenter.read(vm.stateParams.id)
+      .then((data) => {
+        vm.feeCenter = data.feeCenter[0];
+
+        if (data.services) {
+          vm.relatedServices = data.services.length ? 1 : 0;
+          vm.services = data.services;
+        }
+
+        processReference(data.references);
+        vm.setting = true;
+      })
+      .catch(Notify.handleError);
+  }
+
+  function processReference(references) {
+    if (references) {
+      references.forEach((reference) => {
+        if (reference.is_cost) {
+          if (vm.feeCenter.is_principal) {
+            vm.hasCostCenter = 1;
+          } else {
+            vm.isCostCenter = 1;
+            vm.auxiliaryCenter = 1;
+          }
+          vm.feeCenter.is_cost = reference.is_cost;
+          vm.feeCenter.reference_cost_id = reference.account_reference_id;
+          vm.costCenterReference = {
+            account_reference_id : reference.account_reference_id,
+            is_cost : reference.is_cost,
+          };
+        }
+
+        if (!reference.is_cost) {
+          if (vm.feeCenter.is_principal) {
+            vm.hasProfitCenter = 1;
+          } else {
+            vm.isProfitCenter = 1;
+            vm.auxiliaryCenter = 1;
+          }
+
+          vm.feeCenter.is_cost = reference.is_cost;
+          vm.feeCenter.reference_profit_id = reference.account_reference_id;
+          vm.profitCenterReference = {
+            account_reference_id : reference.account_reference_id,
+            is_cost : reference.is_cost,
+          };
+        }
+      });
+    }
+  }
+
+  function auxiliaryFee(value) {
+    vm.auxiliaryCenter = value ? 1 : 0;
+    if (value) {
+      vm.hasProfitCenter = !value;
+      vm.hasCostCenter = !value;
+    }
+
+    if (vm.auxiliaryCenter) {
+      vm.isCostCenter = 0;
+      vm.isProfitCenter = 0;
+    }
```

Comment:
Instead of creating two `if` statement, you could just write one `if` statement with all the code inside : 
```js
if (value) {
  vm.hasProfitCenter = !value;
  vm.hasCostCenter = !value;
  vm.isCostCenter = 0;
  vm.isProfitCenter = 0;
}
```